lib/database_consistency/checkers/validators_fraction_checkers/column_presence_checker.rb

```diff
@@ -34,7 +34,23 @@ def report_template(status, column_name:, error_slug: nil)
       end
 
       def weak_option?
```

Comment:
@djezzzl after refactoring, and considering the edge case you mentioned in the issue, ( and thanks to specs for helping me out), I figured all cases can be covered by simply tweaking the `weak_option?` to make it return false if one of the validators has the `if` condition + `message: :required`  **only when rails 7.1 + the new config is set to false**. 👍🏼